node/node.go

```diff
@@ -344,8 +355,72 @@ var (
 	errNotRightKeySize = errors.New("key received over wire is wrong size")
 	errNoSenderPubKey  = errors.New("no sender public BLS key in message")
 	errWrongShardID    = errors.New("wrong shard id")
+	errInvalidNodeMsg  = errors.New("invalid node message")
+	errIgnoreBeaconMsg = errors.New("ignore beacon sync block")
 )
 
+// validateNodeMessage validate node message
+func (node *Node) validateNodeMessage(ctx context.Context, payload []byte) (
+	[]byte, proto_node.MessageType, error) {
+	atomic.AddUint32(&node.NumTotalNodeMsg, 1)
+
+	// length of payload must > p2pNodeMsgPrefixSize
+
+	// reject huge node messages
+	if len(payload) >= types.MaxEncodedPoolTransactionSize {
+		atomic.AddUint32(&node.NumOversizedMsg, 1)
+		return nil, 0, core.ErrOversizedData
+	}
+
+	// just ignore payload[0], which is MsgCategoryType (consensus/node)
+	msgType := proto_node.MessageType(payload[proto.MessageCategoryBytes])
+
+	switch msgType {
+	case proto_node.Transaction:
+		// nothing much to validate transaction message unless decode the RLP
+		atomic.AddUint32(&node.NumTransactionMsg, 1)
+	case proto_node.Staking:
+		// nothing much to validate staking message unless decode the RLP
+		atomic.AddUint32(&node.NumStakingMsg, 1)
+	case proto_node.Block:
+		switch proto_node.BlockMessageType(payload[p2pNodeMsgPrefixSize]) {
+		case proto_node.Sync:
+			// only non-beacon nodes process the beacon block sync messages
+			if node.Blockchain().ShardID() == shard.BeaconChainShardID ||
+				node.NodeConfig.Role() == nodeconfig.ExplorerNode {
```

Comment:
explorer node also need beacon sync.

Reply:
The original logic is here. https://github.com/harmony-one/harmony/blob/43d0696505927b7586426cc358c4d639bea2c254/node/node_handler.go#L101

explorer node on other shards doesn't participate in beacon block sync.

Reply:
it's likely a bug. Let's fix it here. The explorer node is using state sync to catch up on beacon chain now. With this fix, it will help explorer node to sync beacon chain better

Reply:
okay.